staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json.go

```diff
@@ -32,6 +33,58 @@ import (
 	utilyaml "k8s.io/apimachinery/pkg/util/yaml"
 )
 
+var (
+	// Private copy of jsoniter.ConfigCompatibleWithStandardLibrary to try to shield against possible mutations
+	// from outside. Still does not protect from package level jsoniter.Register*() functions - someone calling them
+	// in some other library will mess with every usage of the jsoniter library in the whole program.
+	// See https://github.com/json-iterator/go/issues/265
```

Comment:
Thanks for the pointer. Is there any thing we can do to prevent misusing package level jsoniter.Register*() functions? Did a quick search in our code base and currently this file seems to be the only place using it. 

Since I see your effort changing use of `encoding/json` to `github.com/json-iterator/go`, does it worth adding some linter/script in CI/makerule that blocks any use of jsoniter.Register*() functions, in case we adopt more and more use of jsoniter?

Reply:
> Is there any thing we can do to prevent misusing package level jsoniter.Register*() functions? Did a quick search in our code base and currently this file seems to be the only place using it.

I think the right thing is to fix the library but the maintainer thinks it is not a problem so... We can have a linting script, as you pointed out. However, I just want to fix the bug and I don't have time to add a linter. Can create a ticket to track this work but I doubt anyone will pick it up :)

Reply:
I agree with you and I see the maintainer's point. It would be great if you could open a ticket to track the linting work.

Reply:
Sure https://github.com/kubernetes/kubernetes/issues/64010